core/src/com/unciv/ui/cityscreen/CityStatsTable.kt

```diff
@@ -52,15 +52,16 @@ class CityStatsTable(val cityScreen: CityScreen): Table() {
                 " " + cityInfo.population.getFreePopulation().toString() + "/" + cityInfo.population.population
 
         var turnsToExpansionString =
-                if (cityInfo.cityStats.currentCityStats.culture > 0) {
+                if (cityInfo.cityStats.currentCityStats.culture > 0 && cityInfo.expansion.chooseNewTileToOwn() != null) {
```

Comment:
Do we need to call `chooseNewTileToOwn()` each time we just calculate a text label?
Why don't use `isAreaMaxed()` instead? Are there any other cases when `chooseNewTileToOwn()` returns null?

Reply:
Sorry - isAreaMaxed means the whole 11-tile wide hexagon is owned by this city as never-to-be-reached upper limit (but for testing I simply set the pool to 30k and waited to turn 84). If that turns out true then the city could only ever grow again if it loses tiles due to conquest. For all other cases the tiles need to be scanned for acquirable ones. That check says _currently_ no expansion - but that could change - conquest of neighboring cities. Or is my thinking way off?

Reply:
Maybe, I misunderstand something, but as I see it: you do a check whether it is currently expanding or not. What can be a reason for not expanding? I see the only 2 reasons: 1) city produces 0 culture (this check is already there), 2) the maximum number of tiles is reached, whole 11-tile wide hexagon is owned.
So my suggestion is to replace a complicated calculations like `chooseNewTileToOwn() != null` with a simple comparison of two integers like `isAreaMaxed()`.

Reply:
Either I am stuck in a mental rut or you are - no offense meant - I know you are cleverer than me - but:
Number of tiles I could expand into = 90 `minus` number I have `minus` number off map `minus` number already owned, whether friendly or not `minus` number I cannot reach because I do not own a neighbor. The easy comparison can only cover "90 - number I have". Off map isn't cached as far as I know, and owned by others can change every turn. So scanning them all to count is needed anyway, and I don't actually need the number just whether it's nonzero, and chooseNewTileToOwn does that nicely. Umm, right?

Reply:
[*] If it turns out I was right then I owe you one for being an insufferable wiseass, or if you are right then I owe you one for a well-earned lesson. Let's see.

Reply:
Ah.. now I understand... you can stuck much earlier than 90 tiles, e.g. if the city is on the edge of the map.
Then you are right, I do not see better solution rather than count all of them - and that's what `chooseNewTileToOwn` does. 👍

Reply:
And now I see a potential bug here (not related to your changes).
Let's imagine a situation: there are two large cities Red (yours) and Green (enemy).
You have found it too close to the opponent, so the start tiles are not all 6, but you quickly bought all others reaching the max 5 range.
![Untitled2](https://user-images.githubusercontent.com/27405436/79381094-1c859980-7f6a-11ea-8633-5ac37f44f51b.png)

Then you capture the Green city and raze it to the ground! 🔥 Yeah, revenge!! 🥳 
Would you get your 2 tiles near the city back? No! 🐞 
Because, `fun chooseNewTileToOwn() ` does the itergations starting from the 2nd row - `for (i in 2..5)` !